hermes-common/src/main/java/pl/allegro/tech/hermes/common/di/factories/ConfigFactoryCreator.java

```diff
@@ -0,0 +1,84 @@
+package pl.allegro.tech.hermes.common.di.factories;
+
+import com.netflix.config.AbstractPollingScheduler;
+import com.netflix.config.ConcurrentCompositeConfiguration;
+import com.netflix.config.ConfigurationManager;
+import com.netflix.config.DynamicConfiguration;
+import com.netflix.config.DynamicPropertyFactory;
+import com.netflix.config.sources.URLConfigurationSource;
+import org.apache.commons.configuration.AbstractConfiguration;
+import org.apache.commons.configuration.EnvironmentConfiguration;
+import org.apache.commons.configuration.SystemConfiguration;
+import org.glassfish.hk2.api.Factory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import pl.allegro.tech.hermes.common.config.ConfigFactory;
+
+import static com.netflix.config.ConfigurationManager.APPLICATION_PROPERTIES;
+import static com.netflix.config.ConfigurationManager.DISABLE_DEFAULT_ENV_CONFIG;
+import static com.netflix.config.ConfigurationManager.DISABLE_DEFAULT_SYS_CONFIG;
+import static com.netflix.config.ConfigurationManager.ENV_CONFIG_NAME;
+import static com.netflix.config.ConfigurationManager.SYS_CONFIG_NAME;
+import static com.netflix.config.ConfigurationManager.URL_CONFIG_NAME;
+
+public class ConfigFactoryCreator implements Factory<ConfigFactory> {
```

Comment:
clever name ;D

Reply:
its not perfect, but it was not easy to get out from this language puzzle ;-)